src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -732,10 +732,26 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                 throw new InvalidOperationException("An 'id' must be specified in the host configuration.");
             }
 
-            JToken watchFiles = (JToken)config["watchFiles"];
-            if (watchFiles != null && watchFiles.Type == JTokenType.Boolean)
+            JToken fileWatchingEnabled = (JToken)config["fileWatchingEnabled"];
+            if (fileWatchingEnabled != null && fileWatchingEnabled.Type == JTokenType.Boolean)
             {
-                scriptConfig.FileWatchingEnabled = (bool)watchFiles;
+                scriptConfig.FileWatchingEnabled = (bool)fileWatchingEnabled;
+            }
+
+            // Configure the set of watched directories, adding the standard built in
+            // set to any the user may have specified
+            if (scriptConfig.WatchDirectories == null)
+            {
+                scriptConfig.WatchDirectories = new HashSet<string>();
```

Comment:
nit: I'd specify `StringComparer.OrdinalIgnoreCase` in the ctor instead of the `.Contains` call below.